File: BoxesDiv2/solve/java/BoxesDiv2Solver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class BoxesDiv2Solver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<Integer> candyCountsBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] candyCounts = new int[candyCountsBoxed.size()];
        for (int _i = 0; _i < candyCountsBoxed.size(); ++_i)
            candyCounts[_i] = candyCountsBoxed.get(_i);
        reader.close();

        BoxesDiv2 solver = new BoxesDiv2();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.findSize(candyCounts));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}

File: BoxesDiv2/solve/cpp/BoxesDiv2Solver.cpp
#include <fstream>
#include <topcoder>
#include "BoxesDiv2.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    vector<int> candyCounts; tc::read(ifs, candyCounts);
    ifs.close();

    std::ofstream ofs(argv[2]);
    BoxesDiv2 solver;
    tc::show(ofs, solver.findSize(candyCounts));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}
